Sources/Fullscreen/FavoriteAdCommentInput/FavoriteAdCommentInputView.swift

```diff
@@ -176,61 +142,35 @@ final class FavoriteAdCommentViewController: UIViewController {
         guard let keyboardInfo = KeyboardNotificationInfo(notification) else { return }
 
         let keyboardVisible = keyboardInfo.action == .willShow
-        let keyboardIntersection = keyboardInfo.keyboardFrameEndIntersectHeight(inView: view)
+        let keyboardIntersection = keyboardInfo.keyboardFrameEndIntersectHeight(inView: self)
 
         UIView.animateAlongsideKeyboard(keyboardInfo: keyboardInfo, animations: { [weak self] in
             guard let self = self else { return }
 
             self.updateScrollViewConstraint(withKeyboardVisible: keyboardVisible, keyboardOffset: keyboardIntersection)
-            self.view.layoutIfNeeded()
-            self.shadowView.updateShadow(using: self.scrollView)
+            self.layoutIfNeeded()
+            self.delegate?.favoriteAdCommentInputView(self, didScroll: self.scrollView)
         })
     }
 
     private func updateScrollViewConstraint(withKeyboardVisible keyboardVisible: Bool, keyboardOffset: CGFloat) {
-        let offset = keyboardOffset + view.windowSafeAreaInsets.bottom
-        scrollViewBottomConstraint.constant = -offset
-    }
-
-    // MARK: - Actions
-
-    @objc private func handleCancelButtonTap() {
-        delegate?.favoriteAdCommentViewControllerDidSelectCancel(self)
-    }
-
-    @objc private func handleSaveButtonTap() {
-        textView.endEditing(true)
-        delegate?.favoriteAdCommentViewController(self, didSelectSaveComment: textView.text)
-    }
-
-    // MARK: - Public methods
-
-    public func startLoading() {
-        loadingView.startAnimating()
-        navigationItem.setRightBarButton(loadingBarButton, animated: true)
-        cancelButton.isEnabled = false
-    }
-
-    public func stopLoading() {
-        loadingView.stopAnimating()
-        navigationItem.setRightBarButton(saveButton, animated: true)
-        cancelButton.isEnabled = true
+        scrollViewBottomConstraint.constant = -keyboardOffset
```

Comment:
I guess this line could be moved into the animation block above, since the rest of this method is removed